reconcile/utils/saasherder.py

```diff
@@ -99,75 +105,73 @@ def _validate_saas_files(self):
+                for e_key, e_value in environment_parameters.items():
+                    if not isinstance(e_value, str):
                         continue
-                    target_parameters = json.loads(target_parameters)
-                    environment_parameters = environment['parameters']
-                    if not environment_parameters:
+                    if '.' not in e_value:
                         continue
-                    environment_parameters = \
-                        json.loads(environment_parameters)
-                    msg = \
-                        f'[{saas_file_name}/{resource_template_name}] ' + \
-                        'parameter found in target ' + \
-                        f'{cluster_name}/{namespace_name} ' + \
-                        f'should be reused from env {environment_name}'
-                    for t_key, t_value in target_parameters.items():
-                        if not isinstance(t_value, str):
-                            continue
-                        # Check for recursivity. Ex: PARAM: "foo.${PARAM}"
-                        replace_pattern = '${' + t_key + '}'
-                        if replace_pattern in t_value:
-                            logging.error(
-                                f'[{saas_file_name}/{resource_template_name}] '
-                                f'recursivity in parameter name and value '
-                                f'found: {t_key}: "{t_value}" - this will '
-                                f'likely not work as expected. Please consider'
-                                f' changing the parameter name')
-                            self.valid = False
-                        for e_key, e_value in environment_parameters.items():
-                            if not isinstance(e_value, str):
-                                continue
-                            if '.' not in e_value:
-                                continue
-                            if e_value not in t_value:
-                                continue
-                            if t_key == e_key and t_value == e_value:
-                                details = \
-                                    f'consider removing {t_key}'
-                            else:
-                                replacement = t_value.replace(
-                                    e_value,
-                                    '${' + e_key + '}'
-                                )
-                                details = \
-                                    f'target: \"{t_key}: {t_value}\". ' + \
-                                    f'env: \"{e_key}: {e_value}\". ' + \
-                                    f'consider \"{t_key}: {replacement}\"'
-                            logging.error(f'{msg}: {details}')
-                            self.valid = False
+                    if e_value not in t_value:
+                        continue
+                    if t_key == e_key and t_value == e_value:
```

Comment:
I tend to find this easier too:
```suggestion
                    if (t_key, t_value) == (e_key, e_value):
```